### PR TITLE
Add stockstrings for reactions in chatlist  & update chatlistitem on reaction event

### LINKED ISSUE #3744: adapt to "reaction summaries"
- set the stock strings [DC_STR_YOU_REACTED](https://c.delta.chat/group__DC__STR.html#ga546c9fed4e379f8ceb41a73957f4834f) and [DC_STR_REACTED_BY](https://c.delta.chat/group__DC__STR.html#gacb5d6d70133eb05aee7e9fd8425a72d0) to [`reaction_by_you` resp. `reaction_by_other`](https://github.com/deltachat/deltachat-android/pull/2985) (may require core update or using core-main, the strings are already on transifex)

- make sure, the chatlist is updated on [DC_EVENT_REACTIONS_CHANGED](https://c.delta.chat/group__DC__EVENT.html#ga613ad73fa800ebd31b66d357e2192f2a)  
([corresponding android pr](https://github.com/deltachat/deltachat-android/pull/2986))

see https://github.com/deltachat/deltachat-core-rust/pull/5387 for details, screenshot and reasoning.

### PR DESCRIPTION
**Pull request by Simon-Laux**
- **add stockstrings for reactions in chatlist**
- **update chatlistitem on reaction event**

closes #3744

depends on a new core version that includes https://github.com/deltachat/deltachat-core-rust/pull/5387 and some other small fix prs (that are included in the next version anyway, so I don't need to list them here, just use core `main` branch)

### CODE REVIEW

**Review by r10s — APPROVED** [review]
thanks a lot! code looks good, did not test though

**Comment by Simon-Laux** [comment]
note to the person merging this: we can not merge it before we have a new core version with [this](https://github.com/deltachat/deltachat-core-rust/pull/5387) (because the missing constants will break the build), maybe that new core version will also include the chatlist events (desktop #3268, core https://github.com/deltachat/deltachat-core-rust/pull/4476) in that case we would drop the event listener commit before merging to resolve the merge conflict.

**Comment by r10s** [comment]
yeah, let's do a core release today, this is also helpful for ios (i am currently afk, will do one unless it happened until i'm back :)

**Review by adzialocha — APPROVED** [review]

**Pull request merged**
